### ISSUE TITLE
bug in UDO parsing

### ISSUE THREAD

**Issue by vlazzarini** [csound/csound#1429]
The following example shows what looks like a bug in overloaded UDO selection by the parser. There
are two UDOs with slightly different input parameters (an extra k-rate input). We try to use the second UDO
with one fewer input, but the parser gets confused.

In the first case (instr 1), the wrong UDO is selected (and the last parameter is set to 0). In the second
case (instr 2), the UDO is selected correctly. 

```
<CsoundSynthesizer>
<CsOptions>
-odac -d
</CsOptions>
<CsInstruments>

0dbfs = 1

opcode Flanger,a,akkk
asig,kfr,kdep,kg xin
printk2 kg
kdep *= 0.005-1/kr
adel delayr  0.005
avd  deltap3 kdep*(0.5+oscili(0.5,kfr))+1/kr
     delayw asig+avd*kg
  xout avd
endop

opcode Flanger,a,akk
 asig,kf,kg xin
 adel delayr  1
 avd  deltapi 1/kf - 1/(2*sr)
     delayw asig+(avd+delay1(avd))*kg/2
  xout avd
endop


instr 1
a1 rand 0.1
a2 Flanger a1*0.25,900,0.9
out a2
endin

instr 2
a1 rand 0.1
a1 *= 0.25
a2 Flanger a1,900,0.9
out a2
endin
</CsInstruments>
<CsScore>
i1 0 1
i2 2 1
</CsScore>
</CsoundSynthesizer>
```

**Comment by kunstmusik** [comment]
I'm going to guess this has to do with the hidden ksmps i-rate override parameter that gets automatically added. I really think we should remove that feature for CS7 to avoid issues like this if that turns out to be the case.

**Comment by vlazzarini** [comment]
yes, that could be the case. The strange thing is that the parser is treating the last parameter as if it were optional, but
only in the first case when there is an inline expression.  

Also if the 3-parameter UDO is commented out, the parser correctly spits out an error.

**Comment by jpffitch** [comment]
Soes ris mean t will not be fixeduntil cs7 ?

**Comment by vlazzarini** [comment]
as @kunstmusik correctly diagnosed, this was to do with the hidden parameter that messes things up. So I have now removed that and the bug is fixed.
